orchagent/vnetorch.cpp

```diff
@@ -1563,12 +1564,36 @@ void VNetRouteOrch::postRouteState(const string& vnet, IpPrefix& ipPrefix, NextH
     fvVector.emplace_back("state", route_state);
 
     state_vnet_rt_tunnel_table_->set(state_db_key, fvVector);
+
+    if (route_state == "active")
```

Comment:
Advertisement must be done only if the VNET attribute is set to [advertise ](https://github.com/Azure/SONiC/blob/a2c68f2b6c40d4b5da164713a988146612ce5cf4/doc/vxlan/Overlay%20ECMP%20with%20BFD.md#212-vnetinterface-table). It should not break the previous behavior of default as no-advertise.

Reply:
Added support for `advertise_prefix` attribute and updated vs tests.